src/arch/x86_64/Emit.zig

```diff
@@ -244,6 +247,25 @@ fn mirPushPop(emit: *Emit, tag: Mir.Inst.Tag, inst: Mir.Inst.Index) InnerError!v
         0b11 => unreachable,
     }
 }
+fn mirPushPopRegsFromCalleePreservedRegs(emit: *Emit, tag: Mir.Inst.Tag, inst: Mir.Inst.Index) InnerError!void {
+    const callee_preserved_regs = bits.callee_preserved_regs;
+    // PUSH/POP reg
+    const opc: u8 = switch (tag) {
+        .push => 0x50,
+        .pop => 0x58,
+        else => unreachable,
+    };
+
+    const regs = emit.mir.instructions.items(.data)[inst].regs_to_push_or_pop;
+    for (callee_preserved_regs) |reg, i| {
+        if ((regs >> @intCast(u5, i)) & 1 == 0) continue;
+        const encoder = try Encoder.init(emit.code, 2);
+        encoder.rex(.{
+            .b = reg.isExtended(),
+        });
+        encoder.opcode_withReg(opc, reg.lowId());
+    }
```

Comment:
I think there may be an issue here: The order in which the registers are pushed to the stack must be the exact opposite of the order in which the registers are popped iirc, i.e. the last register pushed must be the first register popped.

Reply:
this makes a lot of sense. thanks